src/main/java/io/apicurio/perftest/agg/rest/AggregatorResourceImpl.java

```diff
@@ -89,4 +96,44 @@ public String info() {
         return "Hello from Apicurio Performance Test Aggregator";
     }
 
+    /**
+     * @see io.apicurio.perftest.agg.rest.AggregatorResource#workers()
+     */
+    @Override
+    public Set<String> workers() throws Exception {
+        return new HashSet<>(workers);
+    }
+
+    /**
+     * @see io.apicurio.perftest.agg.rest.AggregatorResource#workerStart()
+     */
+    @Override
+    public void workerStart(String workerId) throws Exception {
+        if (workers.add(workerId)) {
+            System.out.println("====> Worker reported is staring id:" + workerId);
+            workersCounter.incrementAndGet();
+        }
+    }
+
+    /**
+     * @see io.apicurio.perftest.agg.rest.AggregatorResource#workerStop()
+     */
+    @Override
+    public void workerStop(String workerId) throws Exception {
+        if (workers.remove(workerId)) {
```

Comment:
Do we need to use a `ConcurrentHashSet` here?

Reply:
yes we need!